// @framer/subtlegradient.kawaii/code/Kawaii.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import {
  Browser,
  CreditCard,
  Ghost,
  IceCream,
  Mug,
  Planet,
  SpeechBubble
} from "react-kawaii";

// Define type of property
interface Props {
  text: string;
  height: number;
  face: string;
  colors: string[];
  mood: string;
}

export class Kawaii extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    face: "Planet",
    colors: []
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    colors: {
      title: "Color",
      type: ControlType.Array,
      maxCount: 1,
      propertyControl: { type: ControlType.Color }
    },
    face: {
      type: ControlType.Enum,
      options: [
        "Planet",
        "Browser",
        "CreditCard",
        "Ghost",
        "IceCream",
        "Mug",
        "SpeechBubble"
      ]
    },
    mood: {
      type: ControlType.Enum,
      options: ["blissful", "sad", "shocked", "happy", "lovestruck", "excited"]
    }
  };

  render() {
    let Comp = Planet;
    switch (this.props.face) {
      case "Browser":
        Comp = Browser;
        break;
      case "CreditCard":
        Comp = CreditCard;
        break;
      case "Ghost":
        Comp = Ghost;
        break;
      case "IceCream":
        Comp = IceCream;
        break;
      case "Mug":
        Comp = Mug;
        break;
      case "Planet":
        Comp = Planet;
        break;
      case "SpeechBubble":
        Comp = SpeechBubble;
        break;
    }

    return (
      <Comp
        size={this.props.height}
        mood={this.props.mood}
        color={this.props.colors[0]}
      />
    );
  }
}
