// @framer/mamezito.bulb-design-system/code/confirmaton.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import { Confirmation } from '@bulb/patterns/modules/Confirmation'

interface Props {
  width: number;
  statusTitle:string;
  statusDescription: string;
  buttonText: string;
}

export class confirmation extends React.Component {
  static defaultProps = {
    statusTitle:"You’ve topped up £14.90",
  statusDescription:"You’ll see the £14.90 in your Bulb account in 3-5 working days.",
    buttonText:"View your payments",

  };

  static propertyControls: PropertyControls<Props> = {
    statusTitle: { type: ControlType.String, title: "Title" },
    statusDescription: { type: ControlType.String, title: "Description "},
    buttonText: { type: ControlType.String, title: "Button text" },
  };

  render() {
    const {
     width, statusTitle,   statusDescription,   buttonText
    } = this.props;
    return (
      <Confirmation
        width={this.props.width}
        statusTitle={statusTitle}
  statusDescription={  statusDescription}
  buttonText={  buttonText}
      
      >
      
      </Confirmation>
    );
  }
}
