// @framer/mamezito.bulb-design-system/code/CTA_Link.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import { CtaLink } from "@bulb/patterns/modules/ButtonsAndLinks/CtaLink";

interface Props {
  text: string;
  width: number;
  height: number;
  fullWidth: boolean;
  purpose: string;
}

export class CTA_Link extends React.Component {
  static defaultProps = {
    text: "Click me",
    fullWidth: false,
    purpose: "primary"
  };

  static propertyControls: PropertyControls<Props> = {
    text: { type: ControlType.String, title: "Title" },
    fullWidth: { type: ControlType.Boolean, title: "Full Width" },
    purpose: {
      type: ControlType.SegmentedEnum,
      title: "Style",
      options: ["primary", "outline", "plain"],
      optionTitles: ["primary", "outline", "plain"]
    }
  };

  render() {
    const { text, fullWidth, width, height, purpose } = this.props;

    return (
      <div
        style={{
          width,
          height,
          display: "flex",
          alignItems: "center",
          justifyContent: "center"
        }}
      >
        <CtaLink
          purpose={purpose || CTA_Link.defaultProps.text}
          fullWidth={fullWidth}
        >
          {text || CTA_Link.defaultProps.text}
        </CtaLink>
      </div>
    );
  }
}
