// @framer/mamezito.bulb-design-system/code/dateInput.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import { DateInput } from "@bulb/patterns/modules/DateInput";

interface Props {
  description:string;
  dateKind:string;
  legend:string;
  yearFormat:string;
  dayValue:string;
  monthValue:string;
  yearValue:string;
  status:string;
  errorMessage:string;
  width: number;
}

export class dateInput extends React.Component {
  static defaultProps = {
    dateKind:"day-month-year",
    legend:"Date of birth",
    yearFormat:"four-digits",
    status: "unknown",
    errorMessage: ""
  };

  static propertyControls: PropertyControls<Props> = {
    legend: { type: ControlType.String, title: "Label" },
    description: { type: ControlType.String, title: "Description" },
    errorMessage: { type: ControlType.String, title: "Error Message" },
    dayValue: { type: ControlType.String, title: "Day value" },
    monthValue: { type: ControlType.String, title: "Month value" },
    yearValue: { type: ControlType.String, title: "Year value" },
    status: { 
        type: ControlType.SegmentedEnum, 
        title: "Status",
        options: ["unknown", "invalid", "valid"],
        optionTitles: ["default", "error", "success"]
      },
      dateKind: { 
        type: ControlType.SegmentedEnum, 
        title: "Type",
        options: ["day-month-year", "month-year"],
        optionTitles: ["DMY", "MY"]
      }
      yearFormat: { 
        type: ControlType.SegmentedEnum, 
        title: "Year format",
        options: ["four-digits", "two-digits"],
        optionTitles: ["four-digits", "two-digits"]
      }
  };

  render() {
    const {
    width,  errorMessage, legend, dateKind, yearFormat, status, dayValue, monthValue, yearValue, description
    } = this.props;
   
    return (
    
        <DateInput
        description={description}
        dateKind={dateKind}
        yearFormat={yearFormat}
          legend={legend}
           width={width}
          errorMessage={errorMessage}
          status={status}
          dayValue={dayValue}
          monthValue={monthValue}
          yearValue={yearValue}
        >
          something
        </DateInput>
    
    );
  }
}
