// @framer/mamezito.bulb-design-system/code/expander.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import { Expander } from "@bulb/patterns/modules/Expanders/Expander";

interface Props {
  width: number;
  height: number;
  hideMessage: string;
  showMessage: string;
  content: string;
  opened: boolean;
}

export class expander extends React.Component {
  static defaultProps = {
    opened: true,
    showMessage:"Show content",
    hideMessage:"Hide content",
    content:"This is an exciting opportunity for a Head of Product to join a very fast growth tech-led company."

  };

  static propertyControls: PropertyControls<Props> = {
    content: { type: ControlType.String, title: "Content" },
    showMessage: { type: ControlType.String, title: "Show message" },
    hideMessage: { type: ControlType.String, title: "Hide message" },
    opened: { type: ControlType.Boolean, title: "Opened" }
  };

  render() {
    const {
      hideMessage,
      showMessage,
      content,
      width,
      height,
      opened
    } = this.props;
    return (
      <Expander
        width={this.props.width}
        showMessage={showMessage}
        hideMessage={hideMessage}
        opened={opened}
      >
      
        {content}
      </Expander>
    );
  }
}
