// @framer/mamezito.bulb-design-system/code/Textinput.tsx
import * as React from "react";
import { PropertyControls, ControlType } from "framer";
import { TextInput } from "@bulb/patterns/modules/TextInput";

interface Props {
  text: string;
  label: string;
  description: string;
  placeholder: string;
  status: string;
  errorMessage: string;
  width: number;
  height: number;
}

export class Textinput extends React.Component {
  static defaultProps = {
    label:  "label",
    description: "description",
    placeholder: "placeholder",
    status: "unknown",
    errorMessage: "field can't be empty"
  };

  static propertyControls: PropertyControls<Props> = {
    label: { type: ControlType.String, title: "Label" },
    description: { type: ControlType.String, title: "Description" },
    placeholder: { type: ControlType.String, title: "Placeholder" },
    errorMessage: { type: ControlType.String, title: "Error Message" }
    status: { 
        type: ControlType.SegmentedEnum, 
        title: "Status",
        options: ["unknown", "invalid", "valid"],
        optionTitles: ["default", "error", "success"]
      }
  };

  render() {
    const {
      label,
      description,
      placeholder,
      status,
      errorMessage, width, height
    } = this.props;
   
    return (
    <div>
        <TextInput
           width={
            this.props.width
          }
          label={label}
          description={description}
          errorMessage={errorMessage}
          placeholder={placeholder}
          status={status}
        >
          something
        </TextInput>
        </div>
    
    );
  }
}
