// @framer/addison.useful-examples/code/Templates/npmTemplate.tsx
/*

    A template for importing components into Framer X. 
    This is assuming you have a sort of design system with an api in place,
    or that you have npm components already made. 

    Included Property Controls in this template:
    -Text
    -Color
    -Boolean
    -Enum

    More information on Property Controls you can use can be found at:
    https://blog.framer.com/property-controls-in-framer-x-bdd550b5010c



    Steps: 

    1: Include this .tsx file in the code folder of your working directory
    2: Import npm package (make sure you've npm installed the package)
    3: Set desired interface props
    4: Expose desired props in propertyControls
    5: Set imported component property to match interface props


    Made by Addison Schultz at Framer

*/

import * as React from "react";
import { PropertyControls, ControlType } from "framer";

// import npm package here
// import { Component } from "npmPackage"

// Define type of property
interface Props {
  text: string;
  color: string;
  boolean: boolean;
  enum: "A" | "B" | "C";
}

export class ComponentTemplate extends React.Component<Props> {
  // Set default properties
  static defaultProps = {
    text: "Framer X",
    color: "#05F",
    boolean: true,
    enum: "A"
  };

  // Items shown in property panel
  static propertyControls: PropertyControls = {
    text: {
      type: ControlType.String,
      title: "Text",
      placeholder: "Framer X"
    },
    color: {
      type: ControlType.Color,
      title: "Color"
    },
    boolean: {
      type: ControlType.Boolean,
      title: "Boolean",
      enabledTitle: "True",
      disabledTitle: "False"
    },
    enum: {
      type: ControlType.Enum,
      options: ["A", "B", "C"],
      optionTitles: ["Option A", "Option B", "Option C"],
      title: "Enum"
    }
  };

  render() {
    return (
      <div>
        {/* 
                    <Component 
                        text={this.props.text}
                        color={this.props.color}
                        disbled={this.props.boolean}
                        type={this.props.enum}
                    /> 
                */}
      </div>
    );
  }
}
