// @framer/addison.useful-examples/code/Overrides/Page.tsx
// This is an override file that can be used to make the page tool work on click
// You can also find a simpler version of this here: https://store.framer.com/package/addison/pagination

import { Data, Override } from "framer";

const data = Data({
  currentPageIndex: 0
});

let totalStories = 0;

// Attach this override to the page component itself
export const Story: Override = props => {
  // Set the total number of stories for later
  totalStories = props.children[0].props.children.length;
  return {
    // Block all events on the Page component
    style: { pointerEvents: "none" },
    // Use the currentPageIndex value to set current page
    currentPage: data.currentPageIndex
  };
};

// Attach this override to the forward arrow / frame
export const TapToAdvance: Override = () => {
  // Increment the currentPageIndex value unless we're at the end
  return {
    opacity: data.currentPageIndex === totalStories - 1 ? 0.3 : 1,
    onTap() {
      if (data.currentPageIndex < totalStories - 1) data.currentPageIndex++;
    }
  };
};

// Attach this override to the backward arrow / frame
export const TapToGoBack: Override = () => {
  return {
    opacity: data.currentPageIndex === 0 ? 0.3 : 1,
    onTap() {
      if (data.currentPageIndex > 0) data.currentPageIndex--;
    }
  };
};

// ** Optional **
// If you want to use dots, you can attach this override to the dots component
// This works best with the dots component in this package: https://store.framer.com/package/framer/examples
export const ChangeDots: Override = props => {
  return {
    currentPage: data.currentPageIndex,
    numberOfPages: totalStories
  };
};
